profiling/src/exporter/mod.rs

```diff
@@ -217,13 +218,30 @@ impl ProfileExporter {
             .iter()
             .chain(files_to_export_unmodified.iter())
             .map(|file| file.name.to_owned())
+            .chain(profile.as_ref().map(|_| "profile.pprof".to_string()))
             .collect();
 
+        let endpoint_counts = profile
+            .as_ref()
+            .map(|p| &p.endpoints_stats)
+            .and_then(|es| es.is_empty().not().then_some(es));
+
+        let start = profile.as_ref().map(|p| {
+            DateTime::<Utc>::from(p.start)
+                .format("%Y-%m-%dT%H:%M:%S%.9fZ")
+                .to_string()
+        });
+        let end = profile.as_ref().map(|p| {
+            DateTime::<Utc>::from(p.end)
+                .format("%Y-%m-%dT%H:%M:%S%.9fZ")
+                .to_string()
+        });
```

Comment:
Hmm I'm concerned about what this is hiding... `start` and `end` are now `Optional<String>` and I don't think this combo makes sense.

Thinking about it, there's two combinations here that make sense:
1. `Profile` can be optional, but if no profile is provided then `start` and `end` MUST be provided
2. `Profile` cannot be optional (and thus there's always a `start` and `end` to use from it)

I slightly favor 1, but I guess we could always add it later as a separate API if we wanted to, so I'm OK with 2. The current mix between them seems dangerous?

Reply:
I do think this is a sharp enough edge that it's a blocker for this PR. A report without `start` and `end` is not a valid report, so it should not be possible to get into such a state.

Reply:
I made profile non-optional